Read information

MKodde opened the pull request:
This is a way to large PR, in effect this contains the addition of:
- database/command test case
- information retrieval command/query/service logic
- setup of the Deprovision client for supporting the GET feature required for the information request.
- tests for several important classes / integrations

mroest commented on src/OpenConext/UserLifecycle/Application/Service/LastLoginService.php:
```diff
@@ -0,0 +1,81 @@
+        $this->logger->debug('Finding the last login entry for the user.');
+        $lastLogin = $this->lastLoginByUserIdQueryHandler->handle($query);
+
+        Assert::notNull(
```
Seems to me that in the case you want to throw an exception here you want to have a more specific NoInformationFoundException ? But, not sure maybe its OK to just return nothing/null?

mroest commented on src/OpenConext/UserLifecycle/Domain/ValueObject/InformationResponse.php:
```diff
@@ -0,0 +1,154 @@
+use JsonSerializable;
+use Webmozart\Assert\Assert;
+
+class InformationResponse implements JsonSerializable
```
I find this interesting. In a general sense I would not expect a response to be a value object. Since the response is unique to the request and can contain specific values. Not sure how this is in this case. I would say the object is immutable and consist of value objects? :)

mroest commented on src/OpenConext/UserLifecycle/Infrastructure/UserLifecycleBundle/Client/DeprovisionClient.php:
```diff
@@ -19,32 +19,170 @@
     public function information(CollabPersonId $user)
     {
-
+        return $this->read(self::DEPROVISION_ENDPOINT, [$user->getCollabPersonId()]);
```
Why not return a reponse object?

MKodde commented:
Thanks again for the review, I've addressed most of your feedback in the final two commits on this branch. Merging, and continuing with the next feature.
